File: 프로그래머스/1/135808. 과일 장수/과일 장수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int k, int m, vector<int> score) {
    int answer = 0;
    
    sort(score.begin(), score.end());
    
    vector<int> tmp;
    while(!score.empty()){
        tmp.push_back(score.back());
        score.pop_back();
        
        if(tmp.size()%m == 0){
            answer += tmp.back()*m;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/135808. 과일 장수/과일 장수.js
function solution(k, m, score) {
    var answer = 0;
    
    score.sort((a, b) => a-b);
    
    while(score.length >= m){
        let box = score.splice(score.length - m, m);
        
        answer += box[0] * m;
    }
    
    return answer;
}